feat(progress-steps): add ProgressStepsContent (#4065)
twilio-labs/paste: packages/paste-website/src/pages/components/progress-steps/index.mdx

### Progress Steps represented vertically

> Use vertical Progress Steps when you’re constrained on horizontal space in your layout.

Reviewer: ```suggestion
You can use vertical Progress Steps when the content of the task is tall and narrow, or you need to add additional content to each step.
```

### Progress Steps represented vertically

> Use vertical Progress Steps when you’re constrained on horizontal space in your layout.

Reviewer 2: **praise:**  woo!

    ProgressStepComplete,
    ProgressStepCurrent,
    ProgressStepError,
  }}
  noInline
>
  {verticalExample}
</LivePreview>

#### Additional content

> Use 'ProgressStepContent' to add additional content to a `orientation="vertical"`, `as="div"` Progress Step.

Reviewer: ```suggestion
Use `ProgressStepContent` to add additional content to a non-interactive vertical Progress Step.
```

> Use 'ProgressStepContent' to add additional content to a `orientation="vertical"`, `as="div"` Progress Step.
> 
> Use additional content when each step requires minimal content or actions—like opening a modal, redirecting, or entering text, and users need to complete steps in a specific sequence, with progress visibly tracked throughout.
> 
> <Callout variant="warning" marginY="space70">
>   <CalloutHeading as="h4">Steps that are completed in another pages and have unique URLs do not automatically track status.</CalloutHeading>

Reviewer: ```suggestion
Use `ProgressStepContent` to add additional content to non-interactive vertical Progress Steps.

Use this when each step requires minimal content or actions, like opening a modal, redirecting, or entering text, and users need to complete steps in a specific sequence, with progress tracked throughout the entire task.

<Callout variant="warning" marginY="space70">
  <CalloutHeading as="h5">Steps that are completed in other pages and have unique URLs do not automatically track status.</CalloutHeading>
```

  <CalloutText>
    Implement the logic locally to ensure the step's status is accurately reflected within the Progress Steps component.
  </CalloutText>
</Callout>

<LivePreview
  scope={{
    ProgressSteps,
    ProgressStepIncomplete,
    ProgressStepComplete,
    ProgressStepCurrent,
    ProgressStepContent,
    Box,
    Avatar,
    Paragraph,
    ButtonGroup,
    Button,
    BusinessIcon,
    Heading
  }}
  noInline
>
  {additionalContentExample}
</LivePreview>

### Internationalization

> Each step within the Progress Step component has a label which can be set to adjust the icon's title for internationalization.

Reviewer: ```suggestion
Each step within the Progress Steps component has a label which can be set to adjust the icon's title for internationalization.
```

  {verticalExample}
</LivePreview>

#### Additional content

> The vertical, non-interactive Progress Steps component has an optional child component that allows you to add additional content.

Reviewer 3: ```suggestion
Use 'ProgressStepContent' to add additional content to a `orientation="vertical"`, `as="div"` Progress Step.
```

> - Use it when all steps can be completed on the same screen, either within the step itself or on another surface (ex: Modal).
> - Use it when the process links to other pages and has a unique URL. but a list of all steps need to be in the same screen.
> - Use it when you need to add additional context to each step.

Reviewer 3: ```suggestion
Use `ProgressStepContent` when:
- All steps can be completed on the same screen, either within the step itself or on another surface (ex: Modal).
- The process links to other pages and has a unique URL. but a list of all steps need to be in the same screen.
- You need to add additional context to each step.
```

Reviewer 2: I updated these to:

Use additional content when each step requires minimal content or actions—like opening a modal, redirecting, or entering text, and users need to complete steps in a specific sequence, with progress visibly tracked throughout.

- If the process does not fit into a mental model of performing tasks, make sure each step is phrased consistently. For example, “Business profile” and “Brand registration.”
- Keep the placement of Progress Steps consistent between different pages (steps).

<Callout variant="neutral" marginY="space70">
  <CalloutText>

>     ProgressStepSeparator is no longer required starting v2.2.1. The separator is now automatically added between each step using CSS. It is backwards compatible, so previous versions will still work.

Reviewer: praise: nice
